src/main/java/hudson/remoting/PingThread.java

```diff
@@ -134,16 +134,16 @@ private void ping() throws IOException, InterruptedException {
      *      and provide a fallback behaviour to be backward compatible with earlier version of remoting library.
      */
     @Deprecated
-    protected abstract void onDead();
+    protected void onDead() {
+        onDead(new RuntimeException("No cause"));
+    }
 
     /**
      * Called when ping failed.
      *
      * @since 2.9
      */
-    protected void onDead(Throwable diagnosis) {
-        onDead();   // fall back
-    }
+    protected abstract void onDead(Throwable diagnosis);
```

Comment:
This is a public class, which could be used somewhere else. Probably you could just add a non-fatal default implementation. E.g.:

```java
protected void onDead(@CheckForNull Throwable diagnosis) {
    System.err.println("Ping failed. Cause: " + diagnosis);
    if (diagnosis != null) {
        diagnosis.printStackTrace(System.err);
    }
}
```